src/tests/payment/test_views.py

```diff
@@ -0,0 +1,228 @@
+from unittest.mock import MagicMock, patch
+
+import pytest
+from django.contrib.auth.models import User
+from django.urls.base import reverse_lazy
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from payment.models import StripeProduct, Subscription
+
+PASSWORD = "password1"
+FIRSTNAME = "John"
+LASTNAME = "Doe"
+EMAIL = "johndoe@example.fr"
+USERNAME = EMAIL
+EMAIL_HOST_USER = "support@ordopro.fr"
+
+
+@pytest.fixture
+def user(db):
+    """Creates and yields a new user."""
+    user = User.objects.create(
+        username=USERNAME, first_name=FIRSTNAME, last_name=LASTNAME, email=EMAIL
+    )
+    user.set_password(PASSWORD)
+    user.save()
+    yield user
+    user.delete()
+
+
+@pytest.fixture
+def staff_user(user):
+    """Creates and yields a staff user."""
+    user.is_staff = True
+    user.save()
+    yield user
+
+
+def authenticate_client_with_token(client, email, password):
+    """Authenticates a client using a token and returns it."""
+    # void previous credentials to avoid "Invalid token" errors
```

Comment:
DRY

Reply:
Now, the conftest.py file handles this and solves the issue